File: src/admin-web/src/Api/manager.js
import request from '../Service/Request';

export const getViewInfo = async (view_name) => {
    return request({
        url: '/api/manager/view-info',
        method: 'post',
        data: { view_name },
    });
}

export const queryView = async (view_name, page, page_size, query) => {
    return request({
        url: '/api/manager/query-view',
        method: 'post',
        data: { view_name, page, page_size, query },
    });
}

export const exportQuery = async (view_name, query) => {
    return request({
        url: '/api/manager/export-query',
        method: 'post',
        responseType: 'blob',
        data: { view_name, query },
    });
}

File: src/api/flaskr/plugins/admin/manager.py
from flask import Flask, request
from flaskr.route.common import make_common_response
from flaskr.service.common.models import raise_param_error
from flaskr.plugins.view.funs import export_query, query_view, get_view


def register_data_manager_route(app: Flask, path_prefix):
    @app.route(path_prefix + "/view-info", methods=["POST"])
    def get_view_route():
        user_id = request.user.user_id
        view_name = request.json.get("view_name")
        return make_common_response(get_view(app, user_id, view_name))

    @app.route(path_prefix + "/query-view", methods=["POST"])
    def query_view_route():
        user_id = request.user.user_id
        view_name = request.get_json().get("view_name")
        if view_name is None:
            raise_param_error("view_name")
        page = request.get_json().get("page", 1)
        page_size = request.get_json().get("page_size", 20)
        query = request.get_json().get("query", {})
        sort = request.get_json().get("sort", [])
        return make_common_response(
            query_view(app, user_id, view_name, page, page_size, query, sort)
        )

    @app.route(path_prefix + "/export-query", methods=["POST"])
    def export_query_route():
        user_id = request.user.user_id
        view_name = request.get_json().get("view_name")
        if view_name is None:
            raise_param_error("view_name")
        query = request.get_json().get("query", {})
        return export_query(app, user_id, view_name, query)

    return app
